src/rebar_prv_common_test.erl

```diff
@@ -218,9 +218,10 @@ select_tests(_, _, _, {error, _} = Error) -> Error;
 select_tests(State, ProjectApps, CmdOpts, CfgOpts) ->
     %% set application env if sys_config argument is provided
     SysConfigs = sys_config_list(CmdOpts, CfgOpts),
-    Configs = lists:flatmap(fun(Filename) ->
-                                rebar_file_utils:consult_config(State, Filename)
-                            end, SysConfigs),
+    Configs = lists:map(fun(Filename) ->
+                            rebar_file_utils:consult_config(State, Filename)
+                        end, SysConfigs),
+    code:add_pathsa(rebar_state:code_paths(State, all_deps)),
```

Comment:
If you add the paths to run the tests, we may want to consider resetting them after the tests. You may also have to look at the bits that handle cover compilation because there could be funny interactions there.

We'd need to double check if this is only about adding paths or also loading modules though, I don't remember at a glance.